stile/hsc/base_tasks.py

```diff
@@ -421,7 +438,10 @@ def computeShapes(self, data, calib, do_shape=True, do_err=True, do_psf=True, do
             localLinearTransform = [wcs.linearizePixelToSky(src.getCentroid()).getLinear()
                                 for src in data]
         if do_shape or do_err:
-            key = data.schema.find("shape.sdss").key
+            if 'galaxy' in mask_type and self.config.do_hsm:  # For any galaxy type, use shape.hsm 
+                key = data.schema.find("shape.hsm.regauss.moments").key  #if available
```

Comment:
`shape.hsm.regauss.moments` is not PSF corrected moments, so we have to use `shape.hsm.regauss.e1` and `shape.hsm.regauss.e2`. For the transformation to sky coordinates, there is the following code in the HSC pipeline, but found that we cannot input |e|>1, so we should write it down by ourselves.

```
# Create an ellipse object using distortion-style ellipticity and an arbitrary radius
# (shape.hsm.regauss.sigma would work too, but it doesn't matter - and
# regauss.sigma *isn't* the PSF-corrected size of the galaxy).
pixel_ellipse = lsst.afw.geom.ellipses.SeparableDistortionDeterminantRadius(
    s.get("shape.hsm.regauss.e1"),
    s.get("shape.hsm.regauss.e2"),
    1.0
)

# Transform the ellipse by the local linear approximation to the WCS:
sky_ellipse = pixel_ellipse.transform(localLinearTransform)

# Extract the e1 and e2 on the sky:
sky_e1 = sky_ellipse.getE1()
sky_e2 = sky_ellipse.getE2()
 ```